### PR TITLE
Refactor aimec

### PR DESCRIPTION
**Pull request by matthiasto**
Splited AIMEC in some smaller atomic functions. Look here for the details : 
https://hackmd.io/LST0LuSqSI2OzYusxP91yg

### CODE REVIEW

File: avaframe/runAna3AIMEC.py [hunk 1]
```diff
@@ -43,17 +43,20 @@
...
-dfa2Aimec.mainDfa2Aimec(avalancheDir, cfgSetup)
+pathDict = dfa2Aimec.mainDfa2Aimec(avalancheDir)
 
-# Extract input file locations
-cfgPath = ana3AIMEC.readAIMECinputs(avalancheDir, dirName='com1DFA')
+# TODO: define referenceFile
+pathDict['numSim'] = len(pathDict['ppr'])
+pathDict['referenceFile'] = 0
+
+pathDict = ana3AIMEC.readAIMECinputs(avalancheDir, pathDict, dirName='com1DFA')
```
Comment (matthiasto) [inline avaframe/runAna3AIMEC.py:52]: Can we read the last argument from the ini file?

**Review by matthiasto — COMMENTED** [review]
The parabola, and hockey are missing the line and point shp file

**Comment by awirb** [comment]
* add option to include in standardTests 
* remove required folder structure

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 2]
```diff
@@ -82,36 +71,115 @@ def readAIMECinputs(avalancheDir, dirName='com1DFA'):
...
+                                              rasterTransfo, interpMethod)
+
+    # assign dem data
+    log.info("Assigning dem data to deskewed raster")
+    newRasterDEM = assignData([cfgPath['demSource']], rasterTransfo,
+                              interpMethod)
+    newRasters['newRasterDEM'] = newRasterDEM[0]
+
+    # Analyze data
+    log.info('Analyzing data in path coordinate system')
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:171]:
```suggestion
    log.debug('Analyzing data in path coordinate system')
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 3]
```diff
@@ -82,36 +71,115 @@ def readAIMECinputs(avalancheDir, dirName='com1DFA'):
...
+    resAnalysis = postProcessAIMECReport(rasterTransfo, pressureLimit, newRasters, cfgPath, cfgFlags)
+
+    # -----------------------------------------------------------
+    # result visualisation + report
+    # -----------------------------------------------------------
+    log.info('Visualisation of results')
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:177]:
```suggestion
    log.info('Visualisation of AIMEC results')
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 4]
```diff
@@ -145,22 +213,37 @@ def mainAIMEC(cfgPath, cfg):
...
                                                  rasterTransfo, interpMethod)
     # assign depth data
     log.info("Assigning depth data to deskewed raster")
-    newRasters['newRasterDepth'] = assignData(cfgPath['depthfileList'],
+    newRasters['newRasterDepth'] = assignData(cfgPath['pfd'],
                                               rasterTransfo, interpMethod)
     # assign speed data
-    if cfgPath['speedfileList']:
+    if cfgPath['pfv']:
         log.info("Assigning speed data to deskewed raster")
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:245]: Lower the log levels to debug

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 5]
```diff
@@ -171,13 +254,18 @@ def mainAIMEC(cfgPath, cfg):
...
     # Analyze data
     log.info('Analyzing data in path coordinate system')
-    resAnalysis = analyzeData(rasterTransfo, pressureLimit, newRasters,
-                              cfgPath, cfgFlags)
+    resAnalysis = postProcessAIMEC(rasterTransfo, pressureLimit, newRasters, cfgPath, cfgFlags)
 
     # -----------------------------------------------------------
     # result visualisation + report
     # -----------------------------------------------------------
     log.info('Visualisation of results')
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:262]:
```suggestion
    log.info('Visualisation of AIMEC results')
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 6]
```diff
@@ -236,36 +324,34 @@ def makeDomainTransfo(cfgPath, cfgSetup, cfgFlags):
...
     rasterTransfo = {}
     rasterTransfo['domainWidth'] = w
     rasterTransfo['xllc'] = xllc
     rasterTransfo['yllc'] = yllc
-    rasterTransfo['cellsize'] = cellsize
+    rasterTransfo['cellsize'] = csz
 
     # read avaPath
-    Avapath = shpConv.readLine(ProfileLayer, DefaultName, sourceData)
+    Avapath = shpConv.readLine(ProfileLayer, DefaultName, dem)
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:352]:
```suggestion
    avaPath = shpConv.readLine(profileLayer, defaultName, dem)
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 7]
```diff
@@ -236,36 +324,34 @@ def makeDomainTransfo(cfgPath, cfgSetup, cfgFlags):
...
     # read split point
-    splitPoint = shpConv.readPoints(cfgPath['splitPointSource'], sourceData)
+    splitPoint = shpConv.readPoints(splitPointSource, dem)
     # add 'z' coordinate to the avaPath
     Avapath, _ = geoTrans.projectOnRaster(dem, Avapath)
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:356]:
```suggestion
    avaPath, _ = geoTrans.projectOnRaster(dem, avaPath)
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 8]
```diff
@@ -430,14 +504,14 @@ def makeTransfoMat(rasterTransfo):
                 new coord system in the cross direction
     """
     w = rasterTransfo['domainWidth']
-    cellsize = rasterTransfo['cellsize']
+    csz = rasterTransfo['cellsize']
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:507]: mh, cellsize should be `cellSize` but I don't know whether this is too much for this PR, maybe open a new issue?

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 9]
```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+    newRasters: dict
+        dictionary containing pressure, velocity and flow depth rasters after
+        transformation
+    cfgPath: dict
+        path to data to analyse
+
+    Returns
+    -------
+    resAnalysis: dict
+        resAnalysis dictionnary containing all results:
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:892]:
```suggestion
        resAnalysis dictionary containing all results:
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 10]
```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
     releaseMass = np.empty((nTopo))
     entrainedMass = np.empty((nTopo))
     finalMass = np.empty((nTopo))
     relativMassDiff = np.empty((nTopo))
+    time = 0
 
-    n = np.shape(lcoord)[0]
-    pCrossAll = np.zeros((nTopo, len(scoord)))
+    log.info('Analyzing mass')
+    log.info('{: <10} {: <10} {: <10}'.format('Sim number ', 'GI ', 'GR '))
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:978]:
```suggestion
    log.debug('Analyzing mass')
    log.debug('{: <10} {: <10} {: <10}'.format('Sim number ', 'GI ', 'GR '))
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 11]
```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+        entMassArray[i] = entMass
+        totalMassArray[i] = totalMass
+        relativMassDiff[i] = (finalMass[i]-finalMass[0])/finalMass[0]*100
+        if not (releaseMass[i] == releaseMass[0]):
+            massDiffers = True
+        log.info('{: <10} {:<10.4f} {:<10.4f}'.format(*[i+1, grIndex[i], grGrad[i]]))
+    if massDiffers:
+        log.warning('Release masses differs between simulations!')
+
+    return releaseMass, entrainedMass, entMassArray, totalMassArray, finalMass, relativMassDiff, grIndex, grGrad, time
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:996]: This is a lot of variables to return, could we combine the mass* ones into one?

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 12]
```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+    log.info('{: <10} {: <10} {: <10}'.format('Sim number ', 'GI ', 'GR '))
     # For each data set
     for i in range(nTopo):
-        rasterdataPres = dataPressure[i]
-        rasterdataDepth = dataDepth[i]
-        rasterdataSpeed = dataSpeed[i]
-        rasterArea = rasterTransfo['rasterArea']
-        # rasterArea[np.where(np.isnan(rasterdataPres))] = np.nan
+        # analyze mass
+        releaseMass[i], entrainedMass[i], finalMass[i], grIndex[i], grGrad[i], entMass, totalMass, time = readWrite(
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:982]: A lot of return values, maybe combine some?

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 13]
```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+    y = rasterTransfo['y']
+
+    # initialize Arrays
+    nTopo = len(PPRCrossMax)
+    runout = np.empty((3, nTopo))
+    runoutMean = np.empty((3, nTopo))
+    elevRel = np.empty((nTopo))
+    deltaH = np.empty((nTopo))
+
+    log.info('Computig Run-Out')
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1052]:
```suggestion
    log.debug('Computig runout')
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 14]
```diff
@@ -837,47 +1084,62 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
-    resAnalysis['relativMassDiff'] = relativMassDiff
-    resAnalysis['growthIndex'] = grIndex
-    resAnalysis['growthGrad'] = grGrad
-    resAnalysis['pCrossAll'] = pCrossAll
-    resAnalysis['pressureLimit'] = pLim
-    resAnalysis['startOfRunoutAngle'] = rasterTransfo['startOfRunoutAngle']
 
-    return resAnalysis
+def analyzeField(rasterTransfo, transformedRasters, dataType):
+    """ Analyse tranformed field A
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1094]:
```suggestion
    """ Analyse transformed field 
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 15]
```diff
@@ -837,47 +1084,62 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+    scoord = rasterTransfo['s']
+    rasterArea = rasterTransfo['rasterArea']
+
+    # initialize Arrays
+    nTopo = len(transformedRasters)
+    maxACrossMax = np.empty((nTopo))
+    ACrossMax = np.zeros((nTopo, len(scoord)))
+    ACrossMean = np.zeros((nTopo, len(scoord)))
+    log.info('Analyzing %s' % (dataType))
+    log.info('{: <10} {: <10}'.format('Sim number ', 'maxCrossMax '))
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1129]: drop to debug

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 16]
```diff
@@ -837,47 +1084,62 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+    # For each data set
+    for i in range(nTopo):
+        rasterData = transformedRasters[i]
+        # rasterArea[np.where(np.isnan(rasterdataPres))] = np.nan
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1133]: remove

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 17]
```diff
@@ -837,47 +1084,62 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
...
+
+        # Max Mean in each Cross-Section for each field
+        maxACrossMax[i], ACrossMax[i], ACrossMean[i] = getMaxMeanValues(rasterData, rasterArea)
+        log.info('{: <10} {:<10.4f}'.format(*[i+1, maxACrossMax[i]]))
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1137]: drop to debug

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 18]
```diff
@@ -890,51 +1152,53 @@ def analyzeArea(rasterTransfo, resAnalysis, pLim, newRasters, cfgPath, cfgFlags)
...
+    refMask = copy.deepcopy(dataPressure[nRef])
     # prepare mask for area resAnalysis
-    newMask = np.where(np.isnan(newMask), 0, newMask)
-    newMask = np.where(newMask < pLim, 0, newMask)
-    newMask = np.where(newMask >= pLim, 1, newMask)
+    refMask = np.where(np.isnan(refMask), 0, refMask)
+    refMask = np.where(refMask < pLim, 0, refMask)
+    refMask = np.where(refMask >= pLim, 1, refMask)
     # comparison rasterdata with mask
     log.info('{: <15} {: <15} {: <15} {: <15} {: <15}'.format(
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1191]: drop to debug
```suggestion
    log.debug('{: <15} {: <15} {: <15} {: <15} {: <15}'.format(
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 19]
```diff
@@ -991,25 +1251,17 @@ def analyzeArea(rasterTransfo, resAnalysis, pLim, newRasters, cfgPath, cfgFlags)
         log.info('{: <15} {:<15.4f} {:<15.4f} {:<15.4f} {:<15.4f}'.format(
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1251]:
```suggestion
        log.debug('{: <15} {:<15.4f} {:<15.4f} {:<15.4f} {:<15.4f}'.format(
```

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 20]
```diff
@@ -1051,10 +1307,10 @@ def readWrite(fname_ent):
 #   growth results
     growthIndex = totMassResults[1]/totMassResults[0]
     growthGrad = (totMassResults[1] - totMassResults[0]) / (timeResults[1] - timeResults[0])
-    return relMass, entMass, finalMass, growthIndex, growthGrad
+    return relMass, entrainedMass, finalMass, growthIndex, growthGrad, entMass, totalMass, time
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1310]: see above, combine the *mass ones

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 21]
```diff
@@ -1066,68 +1322,24 @@ def getMaxMeanValues(rasterdataA, rasterArea, pLim, cInd=None):
...
-        cupper = cInd['cupper']
-        clower = cInd['clower']
-        cupperm = cInd['cupperm']
-        clowerm = cInd['clowerm']
-
-    # Mean max of of a in each Cross-Section
-    ama = np.nansum((aCrossMax*AreaACrossMax)[cupper:clower+1]) / \
-        np.nansum(AreaACrossMax[cupper:clower+1])
-    mma = max(aCrossMax[cupper:clower+1])
+    ACrossMax = np.nanmax(rasterdataA, 1)
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:1341]: AreaSum -> areaSum
ACrossMean -> aCrossMean 
etc

File: avaframe/ana3AIMEC/dfa2Aimec.py [hunk 22]
```diff
@@ -17,59 +17,25 @@
...
-        pfile.write('pathDepoArea=%s,\n' % (emptyVar))
-        pfile.write('pathAOI=%s,\n' % (emptyVar))
-        pfile.write('pathDHM=%s,\n' % (dem[0]))
-        pfile.write('pathEnergy=%s,\n' % (emptyVar))
-        pfile.write('pathVelocity=%s,\n' % (emptyVar))
-        pfile.write('pathResult=%s,\n' % (os.path.join(workDir, 'AimecResults')))
-
-
-def extractMBInfo(avaDir, simName=''):
+def extractMBInfo(avaDir, pathDict, simNameInput=''):
```
Comment (fso42) [inline avaframe/ana3AIMEC/dfa2Aimec.py:20]: Maybe make it more clear that these functions target com1DFA. Either we rename the file to com1DFA2Aimec.py or the functions to something like extractCom1DFAMbInfo...

File: avaframe/out3Plot/outAIMEC.py [hunk 23]
```diff
@@ -29,14 +29,13 @@ def visuTransfo(rasterTransfo, inputData, cfgPath, cfgFlags):
     # Get input data
     # read paths
     projectName = cfgPath['projectName']
-    dirName = cfgPath['dirName']
+    #dirName = cfgPath['dirName']
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:32]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 24]
```diff
@@ -97,33 +96,94 @@ def visuTransfo(rasterTransfo, inputData, cfgPath, cfgFlags):
     ax2.legend(loc=4)
     pU.addColorBar(im, ax2, ticks, pU.cfgPlotUtils['unitppr'])
 
-    outFileName = '_'.join([projectName, dirName, 'DomainTransformation'])
+    #outFileName = '_'.join([projectName, dirName, 'DomainTransformation'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:99]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 25]
```diff
@@ -97,33 +96,94 @@ def visuTransfo(rasterTransfo, inputData, cfgPath, cfgFlags):
...
+    fig, axes = plt.subplots(nrows=1, ncols=3, figsize=(pU.figW*3, pU.figH))
+    fig.subplots_adjust(left=0.05, bottom=0.05, right=0.95, top=0.95, hspace=0.3)
+
+    for ax, dataMax, dataMean, title, unit in zip(axes.flatten(), DataMax, DataMean, Title, Unit):
+        ax.plot(dataMax[0, :], s, '--k', label='Max Reference')
+        ax.plot(dataMean[0, :], s, '-k', label='Mean Reference')
+        ax.plot(dataMax[1, :], s, '--b', label='Max Simulation')
+        ax.plot(dataMean[1, :], s, '-b', label='Mean Simulation')
+
+        ax.set_title(title + 'distribution along the path')
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:149]: Please adjust variable naming: `DataMax` to `dataMax` etc...

File: avaframe/out3Plot/outAIMEC.py [hunk 26]
```diff
@@ -97,33 +96,94 @@ def visuTransfo(rasterTransfo, inputData, cfgPath, cfgFlags):
...
+        ax.legend(loc=4)
+        ax.set_ylabel('s [m]')
+        ax.set_ylim([s.min(), s.max()])
+        ax.set_xlim(auto=True)
+        ax.set_xlabel(unit)
+    pU.putAvaNameOnPlot(ax, cfgPath['projectName'])
+    #outFileName = '_'.join([projectName, dirName, 'plim',
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:156]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 27]
```diff
@@ -162,13 +222,60 @@ def visuRunout(rasterTransfo, resAnalysis, plim, newRasters, cfgPath, cfgFlags):
...
-    ax2.set_ylabel('l [m]')
+    ax2.set_ylabel('s [m]')
     ax2.set_ylim([s.min(), s.max()])
     ax2.set_xlim(auto=True)
     ax2.set_xlabel('$P_{max}(s)$ [kPa]')
 
-    outFileName = '_'.join([projectName, dirName, 'plim',
-                            str(int(plim)), 'slComparison'])
+    #outFileName = '_'.join([projectName, dirName, 'plim',
+    #                        str(int(plim)), 'slComparisonStat'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:231]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 28]
```diff
@@ -162,13 +222,60 @@ def visuRunout(rasterTransfo, resAnalysis, plim, newRasters, cfgPath, cfgFlags):
...
+    for ax, dataMass, title, unit in zip(axes.flatten(), DataMass, Title, Unit):
+        ax.plot(time, dataMass[0, :], '-k', label='Reference')
+        ax.plot(time, dataMass[1, :], '-b', label='Simulation')
+
+        ax.set_title(title + 'mass function of time')
+        ax.legend(loc=4)
+        ax.set_xlabel('t [s]')
+        ax.set_ylabel(unit + '[kg]')
+
+    #outFileName = '_'.join([projectName, dirName, 'massAnalysis'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:276]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 29]
```diff
@@ -162,13 +222,60 @@ def visuRunout(rasterTransfo, resAnalysis, plim, newRasters, cfgPath, cfgFlags):
...
+    totalMassArray = resAnalysis['totalMassArray']
+    time = resAnalysis['time']
+
+    ############################################
+    # prepare for plot
+    Title = ['Entrained ', 'Total ']
+    Unit = ['Entrained Mass ', 'Total Mass ']
+    DataMass = np.array(([None] * 2))
+    DataMass[0] = entMassArray
+    DataMass[1] = totalMassArray
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:259]: adjust variable naming: Title to title etc

File: avaframe/out3Plot/outAIMEC.py [hunk 30]
```diff
@@ -182,18 +289,19 @@ def visuSimple(rasterTransfo, resAnalysis, newRasters, cfgPath, cfgFlags):
     # Get input data
     # read paths
     projectName = cfgPath['projectName']
-    dirName = cfgPath['dirName']
+    #dirName = cfgPath['dirName']
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:292]:
```suggestion
   
```

File: avaframe/out3Plot/outAIMEC.py [hunk 31]
```diff
@@ -230,13 +338,15 @@ def visuSimple(rasterTransfo, resAnalysis, newRasters, cfgPath, cfgFlags):
         pU.addColorBar(im, ax, ticks, unit)
         pU.putAvaNameOnPlot(ax, cfgPath['projectName'])
 
-    outFileName = '_'.join([projectName, dirName,'plim',
+    # outFileName = '_'.join([projectName, dirName,'plim',
+    #                         str(int(plim)), 'referenceFields'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:342]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 32]
```diff
@@ -245,14 +355,15 @@ def visuComparison(rasterTransfo, resAnalysis, inputs, cfgPath, cfgFlags):
     # Get input data
     # read paths
     projectName = cfgPath['projectName']
-    dirName = cfgPath['dirName']
+    #dirName = cfgPath['dirName']
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:358]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 33]
```diff
@@ -296,7 +407,9 @@ def visuComparison(rasterTransfo, resAnalysis, inputs, cfgPath, cfgFlags):
     ax2.set_ylim([s[indStartOfRunout], y_lim])
     plt.subplots_adjust(wspace=0.3)
 
-    outFileName = '_'.join([projectName, dirName, 'plim', str(int(plim)),  str(i), 'comparisonToReference'])
+    #outFileName = '_'.join([projectName, dirName, 'plim', str(int(pLim)),  str(i), 'comparisonToReference'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:410]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 34]
```diff
@@ -359,7 +469,9 @@ def resultWrite(cfgPath, cfgSetup, rasterTransfo, resAnalysis):
                       'pressure_limit: ', str(pressureLimit), ' kPa\n',
                       'start of runout area Angle (SROA angle): ', str(round(startOfRunoutAngle, 2)), ' °\n'])
 
-    outFileName = '_'.join(['Results', projectName, '', dirName, '', 'plim',
+    # outFileName = '_'.join(['Results', projectName, '', dirName, '', 'plim',
+    #                         str(pressureLimit), 'w', str(domainWidth)]) + '.txt'
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:473]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 35]
```diff
@@ -497,16 +605,18 @@ def resultVisu(cfgPath, cfgFlags, rasterTransfo, resAnalysis, plim):
 
     ax1.grid('on')
 
-    outFileName = '_'.join([cfgPath['projectName'], cfgPath['dirName'],
+    #utFileName = '_'.join([cfgPath['projectName'], cfgPath['dirName'],
+    #                    'plim', str(int(plim)), tipo])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:609]: remove

File: avaframe/out3Plot/outAIMEC.py [hunk 36]
```diff
@@ -544,7 +654,9 @@ def resultVisu(cfgPath, cfgFlags, rasterTransfo, resAnalysis, plim):
     plt.ylim([-0.03, 1.03])
     plt.grid('on')
 
-    outFileName = '_'.join([cfgPath['projectName'], cfgPath['dirName'],
+    #outFileName = '_'.join([cfgPath['projectName'], cfgPath['dirName'],
+    #                    'plim', str(int(plim)), 'ROC'])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:658]: remove

File: avaframe/tests/test_dfa2Aimec.py [hunk 37]
```diff
@@ -0,0 +1,62 @@
...
+    # Initialise inputs
+    dirPath = os.path.dirname(__file__)
+    avaTestName = 'avaHockeyChannelPytest'
+    testPath = os.path.join(dirPath, '..', '..', 'benchmarks', avaTestName)
+    pathData = os.path.join(testPath, 'Outputs', 'com1DFA', 'peakFiles')
+    pathDict = dfa2Aimec.mainDfa2Aimec(testPath)
+
+    # get path dictionary for test
+    pathDTest = {}
+    pathDTest['ppr'] = [os.path.join(pathData, 'release1HS_entres_dfa_0.15500_ppr.asc'), os.path.join(pathData, 'release2HS_entres_dfa_0.15500_ppr.asc')]
```
Comment (fso42) [inline avaframe/tests/test_dfa2Aimec.py:26]:
```suggestion
    pathDTest['ppr'] = [os.path.join(pathData, 'release1HS_entres_dfa_0.15500_ppr.asc'), 
                                 os.path.join(pathData, 'release2HS_entres_dfa_0.15500_ppr.asc')]
```
same with the others below

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 38]
```diff
@@ -82,36 +71,115 @@ def readAIMECinputs(avalancheDir, dirName='com1DFA'):
...
     pathResult = os.path.join(avalancheDir, 'Outputs', 'ana3AIMEC', dirName)
     cfgPath['pathResult'] = pathResult
 
     projectName = os.path.basename(avalancheDir)
     cfgPath['projectName'] = projectName
     pathName = os.path.basename(profileLayer[0])
     cfgPath['pathName'] = pathName
-    cfgPath['dirName'] = 'com1DFA'
+    #cfgPath['dirName'] = dirName
+
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:82]: Done

File: avaframe/ana3AIMEC/ana3AIMEC.py (hunk 8 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:507]: made an issue

File: avaframe/ana3AIMEC/ana3AIMEC.py (hunk 11 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:996]: made an issue

File: avaframe/ana3AIMEC/ana3AIMEC.py (hunk 12 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:982]: made an issue

File: avaframe/ana3AIMEC/ana3AIMEC.py (hunk 20 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:1310]: moved to issue

File: avaframe/ana3AIMEC/dfa2Aimec.py (hunk 22 above)
Comment (awirb) [reply avaframe/ana3AIMEC/dfa2Aimec.py:20]: made an issue

File: avaframe/out3Plot/outAIMEC.py (hunk 25 above)
Comment (awirb) [reply avaframe/out3Plot/outAIMEC.py:149]: moved to issue

File: avaframe/out3Plot/outAIMEC.py (hunk 29 above)
Comment (awirb) [reply avaframe/out3Plot/outAIMEC.py:259]: moved to issue

File: avaframe/ana3AIMEC/ana3AIMEC.py (hunk 21 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMEC.py:1340]: Done

File: avaframe/ana3AIMEC/dfa2Aimec.py (hunk 22 above)
Comment (awirb) [reply avaframe/ana3AIMEC/dfa2Aimec.py]: Done

**Pull request merged**